### PR TITLE
(on hold) Read Strip URI from Ingress Annotations

### PR DESCRIPTION
**Pull request by paniagua**
[Strip URI](https://getkong.org/docs/0.10.x/proxy/) It may be desirable to specify a URI prefix to match an API,
but not include it in the upstream request.

It means that if the request is '/omnicall/path' it removes the
'/omnicall' piece, which is not desired.

The default value is `true`. This PR reads the value from the
ingress annotations. E.g.

```
annotations:
  ingress.kubernetes.io/strip-uri: "[true|false]"
```

I'll add a test and make a PR to the upstream

### CODE REVIEW

File: docs/examples/install.yaml [hunk 1]
```diff
@@ -13,11 +13,11 @@ spec:
       terminationGracePeriodSeconds: 60
       containers:
       - name: koli-controller
-        image: 'quay.io/koli/kong-ingress:v0.2.1-alpha'
+        image: 'salemove/kong-ingress'
```
Comment (indrekj) [inline docs/examples/install.yaml]: lets not change the example, will cause conflicts when updating to a new version

File: pkg/controller/controller.go [hunk 2]
```diff
@@ -313,10 +314,17 @@ func (k *KongController) syncIngress(key string, numRequeues int) error {
...
+      			if err != nil {
+        			stripUri = false
+        			glog.Infof("Failed to parse strip-uri annotation, setting it to false")
+      			}
+
 			apiBody := &kong.API{
 				Name:        apiName,
 				Hosts:       []string{r.Host},
 				UpstreamURL: upstreamURL,
+        			StripUri:   stripUri,
```
Comment (indrekj) [inline pkg/controller/controller.go]: indentation

File: versioning.mk [hunk 3]
```diff
@@ -3,15 +3,15 @@ VERSION ?= unknown
...
 
-IMAGE := ${KOLI_REGISTRY}${IMAGE_PREFIX}/${SHORT_NAME}:${VERSION}
-MUTABLE_IMAGE := ${KOLI_REGISTRY}${IMAGE_PREFIX}/${SHORT_NAME}:${MUTABLE_VERSION}
+IMAGE := ${REGISTRY}${IMAGE_PREFIX}/${SHORT_NAME}:${VERSION}
+MUTABLE_IMAGE := ${REGISTRY}${IMAGE_PREFIX}/${SHORT_NAME}:${MUTABLE_VERSION}
 
 info:
 	@echo "Build tag:       ${VERSION}"
-	@echo "Registry:        ${KOLI_REGISTRY}"
+	@echo "Registry:        ${REGISTRY}"
```
Comment (indrekj) [inline versioning.mk]: I don't think you need changes to `versioning.mk` because you use manual `docker push` to update the image

Comment (paniagua) [reply versioning.mk]: This is included in the `Makefile`, `Makefile` references `IMAGE`

**Comment by paniagua** [comment]
Made a PR to the main repo. waiting for feedback until the end of the week.

**Comment by paniagua** [comment]
We are working on getting these changes approved/added in the upstream directly.

**Pull request closed without merging**